### PR TITLE
[Chore]: Technical: add types for processors

### PR DESCRIPTION
**Pull request by dariaterekhovaae**
Signed-off-by: Daria Terekhova <daria.terekhova@actionengine.com>

### CODE REVIEW

File: src/processors/data-processor.ts [hunk 1]
```diff
@@ -58,20 +63,20 @@ const IGNORE_DATA_TYPES = Object.keys(AnalyzerDATA_TYPES).filter(
 
 export const PARSE_FIELD_VALUE_FROM_STRING = {
   [ALL_FIELD_TYPES.boolean]: {
-    valid: d => typeof d === 'boolean',
-    parse: d => d === 'true' || d === 'True' || d === 'TRUE' || d === '1'
+    valid: (d: any) => typeof d === 'boolean',
```
Comment (heshan0131) [inline src/processors/data-processor.ts:66]: `(d: unknown): boolean => typeof d === 'boolean'`

File: src/processors/data-processor.ts [hunk 2]
```diff
@@ -58,20 +63,20 @@ const IGNORE_DATA_TYPES = Object.keys(AnalyzerDATA_TYPES).filter(
...
+    parse: (d: any) => d === 'true' || d === 'True' || d === 'TRUE' || d === '1'
```
Comment (heshan0131) [inline src/processors/data-processor.ts:67]: `(d: unknown): boolean`

File: src/processors/data-processor.ts [hunk 3]
```diff
@@ -58,20 +63,20 @@ const IGNORE_DATA_TYPES = Object.keys(AnalyzerDATA_TYPES).filter(
...
   },
   [ALL_FIELD_TYPES.integer]: {
-    valid: d => parseInt(d, 10) === d,
-    parse: d => parseInt(d, 10)
+    valid: (d: any) => parseInt(d, 10) === d,
```
Comment (heshan0131) [inline src/processors/data-processor.ts:70]: same below, use `(d: unknown): boolean` for validate

File: src/processors/data-processor.ts [hunk 4]
```diff
@@ -58,20 +63,20 @@ const IGNORE_DATA_TYPES = Object.keys(AnalyzerDATA_TYPES).filter(
...
+    parse: (d: any) => parseInt(d, 10)
```
Comment (heshan0131) [inline src/processors/data-processor.ts:71]: `(value: unknown): number`

File: src/processors/data-processor.ts [hunk 5]
```diff
@@ -159,10 +164,16 @@ export function parseRowsByFields(rows, fields) {
...
- * @type {typeof import('./data-processor').getSampleForTypeAnalyze}
  */
-export function getSampleForTypeAnalyze({fields, rows, sampleCount = 50}) {
+export function getSampleForTypeAnalyze({
+  fields,
+  rows,
+  sampleCount = 50
+}: {
+  fields: string[];
+  rows: any[][];
```
Comment (heshan0131) [inline src/processors/data-processor.ts:174]: `rows: unknown[][]`

File: src/processors/data-processor.ts [hunk 6]
```diff
@@ -221,9 +232,13 @@ function cleanUpFalsyCsvValue(rows) {
  * @param geoFieldIdx field index
  * @param field
  * @param i
- * @type {typeof import('./data-processor').parseCsvRowsByFieldType}
  */
-export function parseCsvRowsByFieldType(rows, geoFieldIdx, field, i) {
+export function parseCsvRowsByFieldType(
+  rows: any[][],
```
Comment (heshan0131) [inline src/processors/data-processor.ts:237]: `rows: unknown[][]`

File: src/processors/data-processor.ts [hunk 7]
```diff
@@ -630,11 +641,15 @@ export function validateInputData(data) {
   return {fields: updatedFields, rows};
 }
 
-function findNonEmptyRowsAtField(rows, fieldIdx, total) {
-  const sample = [];
+function findNonEmptyRowsAtField(
+  rows: Record<number, unknown>[],
```
Comment (heshan0131) [inline src/processors/data-processor.ts:645]: `rows: unknown[][]`

File: src/processors/file-handler.ts [hunk 8]
```diff
@@ -47,25 +50,25 @@ const JSON_LOADER_OPTIONS = {
   ]
 };
 
-export function isGeoJson(json) {
+export function isGeoJson(json: any): boolean {
```
Comment (heshan0131) [inline src/processors/file-handler.ts:53]: you can use type guard here
`isGeoJson(json: unknown): json is Feature | FeatureCollection`

File: src/processors/file-handler.ts [hunk 9]
```diff
@@ -47,25 +50,25 @@ const JSON_LOADER_OPTIONS = {
...
   // json can be feature collection
   // or single feature
   return isPlainObject(json) && (isFeature(json) || isFeatureCollection(json));
 }
 
-export function isFeature(json) {
+export function isFeature(json: any): boolean {
```
Comment (heshan0131) [inline src/processors/file-handler.ts:59]: ` isFeature(json: unknown): json is Feature`

File: src/processors/file-handler.ts [hunk 10]
```diff
@@ -47,25 +50,25 @@ const JSON_LOADER_OPTIONS = {
   ]
 };
 
-export function isGeoJson(json) {
+export function isGeoJson(json: any): boolean {
   // json can be feature collection
   // or single feature
   return isPlainObject(json) && (isFeature(json) || isFeatureCollection(json));
```
Comment (heshan0131) [inline src/processors/file-handler.ts:56]: add type `isPlainObject(obj: unknown): obj is Record<string, unknown>`

File: src/processors/file-handler.ts [hunk 11]
```diff
@@ -47,25 +50,25 @@ const JSON_LOADER_OPTIONS = {
...
   return json.type === 'FeatureCollection' && json.features;
 }
 
-export function isRowObject(json) {
+export function isRowObject(json: any): boolean {
   return Array.isArray(json) && isPlainObject(json[0]);
 }
 
-export function isKeplerGlMap(json) {
+export function isKeplerGlMap(json: any): Boolean {
```
Comment (heshan0131) [inline src/processors/file-handler.ts:71]:
```
type ValidKeplerGlMap = {
datasets: unknown
config: unknown
info: Record<string, string>
}
```

`isKeplerGlMap(json: unknown): json is ValidKeplerGlMap`

File: src/processors/data-processor.ts [hunk 12]
```diff
@@ -58,20 +63,22 @@ const IGNORE_DATA_TYPES = Object.keys(AnalyzerDATA_TYPES).filter(
...
-    valid: d => parseInt(d, 10) === d,
-    parse: d => parseInt(d, 10)
+    // @ts-ignore
+    valid: (d: unknown): boolean => parseInt(d, 10) === d,
+    // @ts-ignore
+    parse: (d: unknown): number => parseInt(d, 10)
   },
   [ALL_FIELD_TYPES.timestamp]: {
-    valid: (d, field) =>
+    valid: (d: any, field: Field) =>
```
Comment (heshan0131) [inline src/processors/data-processor.ts:76]: `valid: (d: unkonwn, field: Field): boolean;`

File: src/processors/data-processor.ts [hunk 13]
```diff
@@ -58,20 +63,22 @@ const IGNORE_DATA_TYPES = Object.keys(AnalyzerDATA_TYPES).filter(
...
       ['x', 'X'].includes(field.format) ? typeof d === 'number' : typeof d === 'string',
-    parse: (d, field) => (['x', 'X'].includes(field.format) ? Number(d) : d)
+    parse: (d: any, field: Field) => (['x', 'X'].includes(field.format) ? Number(d) : d)
   },
   [ALL_FIELD_TYPES.real]: {
-    valid: d => parseFloat(d) === d,
+    valid: (d: any) => parseFloat(d) === d,
```
Comment (heshan0131) [inline src/processors/data-processor.ts:81]: `(d: unknown): boolean`

**Review by heshan0131 — APPROVED** [review]
Please address the last 2 comments and merge

File: src/processors/data-processor.ts (hunk 1 above)
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

File: src/processors/data-processor.ts (hunk 2 above)
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

File: src/processors/data-processor.ts (hunk 3 above)
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

File: src/processors/data-processor.ts (hunk 4 above)
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

File: src/processors/data-processor.ts (hunk 5 above)
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

File: src/processors/data-processor.ts [hunk 14]
```diff
@@ -197,9 +208,9 @@ export function getSampleForTypeAnalyze({fields, rows, sampleCount = 50}) {
  * Convert falsy value in csv including `'', 'null', 'NULL', 'Null', 'NaN'` to `null`,
  * so that type-analyzer won't detect it as string
  *
- * @param {Array<Array>} rows
+ * @param rows
  */
-function cleanUpFalsyCsvValue(rows) {
+function cleanUpFalsyCsvValue(rows: any[][]): void {
```
[the remark this answers is not in the document]
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

File: src/processors/data-processor.ts (hunk 6 above)
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

File: src/processors/data-processor.ts (hunk 7 above)
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

File: src/processors/file-handler.ts (hunk 8 above)
Comment (jagerts) [reply src/processors/file-handler.ts]: fixed

File: src/processors/file-handler.ts (hunk 9 above)
Comment (jagerts) [reply src/processors/file-handler.ts]: fixed

File: src/processors/file-handler.ts [hunk 15]
```diff
@@ -47,25 +50,25 @@ const JSON_LOADER_OPTIONS = {
...
   return isPlainObject(json) && (isFeature(json) || isFeatureCollection(json));
 }
 
-export function isFeature(json) {
+export function isFeature(json: any): boolean {
   return json.type === 'Feature' && json.geometry;
 }
 
-export function isFeatureCollection(json) {
+export function isFeatureCollection(json: any): boolean {
```
[the remark this answers is not in the document]
Comment (jagerts) [reply src/processors/file-handler.ts]: fixed

File: src/processors/file-handler.ts (hunk 10 above)
Comment (jagerts) [reply src/processors/file-handler.ts:58]: added

File: src/processors/file-handler.ts (hunk 11 above)
Comment (jagerts) [reply src/processors/file-handler.ts]: added

File: src/processors/data-processor.ts (hunk 12 above)
Comment (jagerts) [reply src/processors/data-processor.ts]: fixed

**Pull request merged**